src/pages/Main/components/hooks/useFetchSearchPosts.ts

```diff
@@ -0,0 +1,16 @@
+import { keepPreviousData, useQuery } from '@tanstack/react-query';
+import { getPostListBySearch } from '@/apis/getPosts';
+
+interface UseFetchSearchPostsParams {
+  search?: string;
+}
+
+const useFetchSearchPosts = ({ search }: UseFetchSearchPostsParams) =>
```

Comment:
api 호출코드에서도 같이수정해야 할 것 같은데 search 하나라면 객체로 interface만들어서 타입정의해주는것보다 인라인으로 가는게 더 가독성이 좋을 것 같습니다. 

```suggestion
const useFetchSearchPosts = (search : string) =>
```

Reply:
그럴것 같네요!!
수정해보겠습니다

Reply:
반영하시고나면 resolve해주세요 :)